src/pybind/mgr/devicehealth/module.py

```diff
@@ -105,23 +105,21 @@ def handle_command(self, _, cmd):
         self.log.error("handle_command")
 
         if cmd['prefix'] == 'device query-daemon-health-metrics':
-            who = cmd.get('who', '')
-            if who[0:4] != 'osd.':
-                return -errno.EINVAL, '', 'not a valid <osd.NNN> id'
-            osd_id = who[4:]
+            (daemon_type, daemon_id) = cmd.get('who', '').split('.')
```

Comment:
Need some error handling here to give a friendly message on a 'who' that doesn't contain the period